# 01 · Understanding the Titanic data

Question: which recorded passenger characteristics are associated with survival? This notebook inspects the original data without changing it.

In [1]:
from pathlib import Path
import pandas as pd

root = Path.cwd().parent if Path.cwd().name == "notebooks" else Path.cwd()
raw = pd.read_csv(root / "data" / "Titanic-Dataset.csv")

In [2]:
raw.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [3]:
print(f"Rows: {raw.shape[0]} | Columns: {raw.shape[1]}")

Rows: 891 | Columns: 12


In [4]:
raw.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name               str
Sex                str
Age            float64
SibSp            int64
Parch            int64
Ticket             str
Fare           float64
Cabin              str
Embarked           str
dtype: object

In [5]:
print("Numerical: Age, Fare, SibSp, Parch")
print("Categorical: Sex, Pclass (ordinal), Embarked, Cabin")
print("Identifiers / text: PassengerId, Name, Ticket")
print("Target: Survived (0 = did not survive, 1 = survived)")

Numerical: Age, Fare, SibSp, Parch
Categorical: Sex, Pclass (ordinal), Embarked, Cabin
Identifiers / text: PassengerId, Name, Ticket
Target: Survived (0 = did not survive, 1 = survived)


The supplied file contains 891 passenger records, not everyone aboard the ship. PassengerId is an identifier, not a meaningful survival factor.